web/src/pages/Settings/Streams/StreamModal/StreamModal.jsx

```diff
@@ -190,11 +191,16 @@ const StreamModal = ({ stream, onClose, apply, del }) => {
                     streamTemplate.fields.map((field) => {
                         switch (field.type) {
                         case "text":
+                            let type = "text";
+                            if(field.name == "password"){
+                                type = "password";
```

Comment:
i think i'd probably prefer to change the `type` of the password fields, instead of behaving magically with specifically named `text` fields. this also creates a new component - it's perhaps somewhat more LOC, but more maintainable, and you don't have to change the interface of some underlying component.